mlx_lm/generate.py

```diff
@@ -322,15 +352,16 @@ def generate_step(
         kv_group_size (int): Group size for KV cache quantization. Default: ``64``.
         quantized_kv_start (int): Step to begin using a quantized KV cache.
            when ``kv_bits`` is non-None. Default: ``0``.
-        prompt_prorgress_callback (Callable[int, int]): A call-back which takes the
+        prompt_progress_callback (Callable[int, int]): A call-back which takes the
            prompt tokens processed so far and the total number of prompt tokens.
+        search_strategy (SearchStrategy): The search strategy to use. This defaults to
+            `LinearSearch`.
 
     Yields:
-        Tuple[mx.array, mx.array]: One token and a vector of log probabilities.
+        Tuple[int, mx.array]: One token and a vector of log probabilities.
```

Comment:
Not related to the main change, but I think the typing was wrong here. I'm pretty sure this method generates int, as it calls `.item()` on the token array, but I might be missing something.